Fix parsing of paths for unmask

ver4a opened the pull request:
This fixes an issue where multiple paths separated by a colon were treated as a single path, contrary to what docs say and unlike how mask option works.

<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).

Finally, be sure to sign commits with your real name. Since by opening
a PR you already have commits, you can add signatures if needed with
something like `git commit -s --amend`.
-->

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
Fixed a bug where multiple paths could not be set in --security-opt=unmask
```

Luap99 reviewed (commented):
Thanks, please add a test for it.

Overall I hate such string.Split() calls for paths because it means all users will be unable to specify paths with a colon in them. Given it is is already done with mask I guess it is good to be consistent so I am fine with it.

ver4a commented:
I'll try, but I'm not exactly sure what's going on. The tests currently test for this
```
podman run -d --name=maskCtr2 --security-opt unmask=/proc/acpi:/sys/firmware
```
and that does work, both paths are masked by default. But when I try adding paths that are read-only by default, it doesn't effect them. I don't understand how the parsing changes that in any way.
Without this PR `/proc/acpi` and `/sys/firmware` get unmasked, `/proc/sys` and `/sys/fs/cgroup` remain read-only:
```
unmask=/proc/sys:/sys/fs/cgroup:/proc/acpi:/sys/firmware
```

I don't understand this behavior, I guess there could be two tests, one for masked paths and one for read-only paths, but it's strange. I'll double check this later when I have more time.

ver4a commented:
I still don't know why some paths worked, but I've added a simple test with one that I can reproduce doesn't.

> Overall I hate such string.Split() calls for paths because it means all users will be unable to specify paths with a colon in them. Given it is is already done with mask I guess it is good to be consistent so I am fine with it.

Agree, it's not pretty, I did it this way for consistency and also because I know absolutely no GO :)

giuseppe commented on test/e2e/run_test.go:
```diff
@@ -423,6 +423,9 @@ var _ = Describe("Podman run", func() {
 		session.WaitWithDefaultTimeout()
 		Expect(session.OutputToString()).To(Not(BeEmpty()))
 		Expect(session).Should(ExitCleanly())
+		session = podmanTest.Podman([]string{"exec", "maskCtr2", "mkdir", "/sys/fs/cgroup/testcgroup"})
```
this won't work with rootless and cgroupfs because the user doesn't get a delegated cgroup.

It would be better to run something like `grep /sys/fs/cgroup /proc/self/mountinfo` and make sure there is `rw` in the mount flags.

giuseppe reviewed (commented):
could you please use your real name to sign off the commit?

mheon commented:
LGTM once comment is addressed

baude commented:
@ver4a would you be willing to push this over the finish line?  much appreciated if so.

ver4a commented:
Sorry for the delay, I got distracted with other stuff.

Could you please take another look at this @giuseppe? I've signed-off the commit with my legal name and made a new test based on your suggestion.

giuseppe reviewed (approved):
LGTM

Luap99 reviewed (approved):
/lgtm
